docs/data/data-grid/ai-assistant/ai-assistant.md

```diff
@@ -0,0 +1,163 @@
+Type the prompt like "sort by name" or "show amounts larger than 1000" in the prompt input field, and the Data Grid will update accordingly.
+
+To enable client-side of this feature, pass `enableAiAssistant` prop.
+
+:::success
+In supported browsers, the prompt can be entered using voice.
+:::
+
+To increase the accuracy of the language processing, provide example values for the available columns.
+This can be done in the following ways.
+
+:::info
+AI assistant demos use a utility function `mockPromptResolver()` to simulate the API that resolves user's prompts.
+In a real-world scenario, replace this with [MUI's](/x/react-data-grid/ai-assistant/#with-muis-service) or [your own](/x/react-data-grid/ai-assistant/#with-custom-service) processing service.
+
+`mockPromptResolver()` can handle a predefined set of prompts:
+
+- `sort by name`
+- `sort by company name and employee name`
+- `show people from the EU`
+- `order companies by amount of people`
+
+You can use suggestions to quickly enter prompts that are supported by the mock resolver.
+
+:::
+
+## Custom examples
+
+You can provide custom examples as a context for the prompt processing through the `examples` prop in the `columns` array.
+The `examples` prop should contain an array of possible values for that column.
+
+{{"demo": "AssistantWithExamples.js", "bg": "inline"}}
+
+## Use row data for examples
+
+Pass `allowAiAssistantDataSampling` prop to allow use of the row data to generate column examples.
+This is useful if you are dealing with non-sensitive data and want to skip creating custom examples for each column.
+
+Data is collected randomly on the cell level, which means that the examples per column might not come from the same rows.
+
+For `aiAssistant.getPromptContext()` API method, pass `allowDataSampling` flag as a parameter.
+
+```ts
+const context = React.useMemo(
+  () => apiRef.current.aiAssistant.getPromptContext(allowDataSampling),
+  [apiRef, allowDataSampling],
+);
+```
+
+{{"demo": "AssistantWithDataSampling.js", "bg": "inline"}}
+
+## Using Server-side data
+
+An example of combining prompt control with the [Server-side data](/x/react-data-grid/server-side-data/)
+
+{{"demo": "AssistantWithDataSource.js", "bg": "inline"}}
+
+## Processing service integration
+
+Natural language prompts must be processed by a service to understand what kind of state changes must be applied to the Data Grid to match the user's request.
+You can use MUI's processing service or build it by yourself.
+
+### With MUI's service
+
+Data Grid provides all necessary pieces to make the service integration easy.
+
+1. Enable the AI Assistant feature by adding `enableAiAssistant` prop.
+   A new toolbar button will appear in the default [`<Toolbar />`](/x/react-data-grid/components/toolbar/).
+   This button opens `<AssistantPanel />` that can take user's prompts.
+2. Contact [sales@mui.com](mailto:sales@mui.com) to get an API key for our processing service.
```

Comment:
Maybe this should be step 1? Seems like you'd want to take care of this before all else.